src/browser-polyfill.js

```diff
@@ -132,7 +132,48 @@ if (typeof browser === "undefined") {
         }
 
         return new Promise((resolve, reject) => {
-          target[name](...args, makeCallback({resolve, reject}, metadata));
+          function callAPIWithNoCallback() {
+            try {
+              target[name](...args);
+              if (chrome.runtime.lastError) {
```

Comment:
This `chrome.runtime.lastError` check seems misplaced. The `target[name](...args)` call is asynchronous, so the error will never be set to that.

Worse, with the current code, if someone tries to use call e.g. `browser.pageAction.show` inside a function where `lastError` is set, then the promise will always be rejected.

Reply:
:+1: it was already supposed to be removed and I forgot, thanks for pointing it out.